### PR TITLE
Set OAuth client ID and client secret at compile time

### PR DESCRIPTION
**Pull request by chanseokoh**
For #342.

Set client ID and client secret at compile-time by auto-generating `Constants.java` with `templating-maven-plugin`. (The same mechanism as in the `.eclipse.usagetracker` bundle.)

### CODE REVIEW

**Comment by briandealwis** [comment]
lgtm

File: plugins/com.google.cloud.tools.eclipse.appengine.login/java-templates/com/google/cloud/tools/eclipse/appengine/login/Constants.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.google.cloud.tools.eclipse.appengine.login;
+
+/**
+ * Placeholder constants initialized at compile-time.
+ */
+public class Constants {
+
+  public static final String OAUTH_CLIENT_ID = "@oauth.client.id@";
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.login/java-templates/com/google/cloud/tools/eclipse/appengine/login/Constants.java]: How are these set? Where do the values actually come from?
Please make sure our docs are updated to include this information.

File: plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginTemporaryTester.java [hunk 2]
```diff
@@ -54,8 +54,8 @@ private File getCredentialFile() throws IOException {
     }
 
     Map<String, String> credentialMap = new HashMap<>();
-    credentialMap.put(CLIENT_ID_LABEL, GoogleLoginService.OAUTH_CLIENT_ID);
-    credentialMap.put(CLIENT_SECRET_LABEL, GoogleLoginService.OAUTH_CLIENT_SECRET);
+    credentialMap.put(CLIENT_ID_LABEL, Constants.OAUTH_CLIENT_ID);
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginTemporaryTester.java]: It's useful to support retrieving these values from properties or the environment for development.  See [GPE](https://github.com/GoogleCloudPlatform/google-plugin-for-eclipse/blob/master/plugins/com.google.gdt.eclipse.login/src/com/google/gdt/eclipse/login/GoogleLogin.java#L80) for an example

**Comment by chanseokoh** [comment]
This is what I added in `README.md`:

# Release

## Compile-time Constant Injection

We plug a few constants into code at compile-time.  We do this to avoid
hard-coding these values and hamper their public visibilty.

| Constant                     | Bundle        | System Property       |
| ---------------------------- | ------------- | --------------------- |
| OAuth client ID              | login         | `oauth.client.id`     |
| OAuth client secert          | login         | `oauth.client.secret` |
| Google Analytics Tracking ID | usagetracking | `ga.tracking.id`      |

The Java classes that hold these constant values are auto-generated
from template Java source files, using [templating-maven-plugin]
(http://www.mojohaus.org/templating-maven-plugin/).  The values are
populated from system properties in the auto-gen process. That is, for
example, `mvn -Doauth.client.id=id1234 templating:filter-sources` or
`mvn -Doauth.client.id=id1234 package` will result in injecting `id1234`
as a OAuth client ID.

### Injecting Constants for Debugging

You can also easily inject values for an Eclipse debug session, since Eclipse
auto-generates the Java classes (if configured to do auto-build).  Providing
relevant system properties when launching a dev instance of Eclipse and deleting
a file to re-generate will do the job.

File: README.md [hunk 3]
```diff
@@ -236,4 +236,30 @@ the identifiers are not p2 identifiers, and so features do not
...
 
+## Compile-time Constant Injection
+
+We plug a few constants into code at compile-time.  We do this to avoid
+hard-coding these values and hamper their public visibilty.
+
+| Constant                     | Bundle        | System Property       |
+| ---------------------------- | ------------- | --------------------- |
+| OAuth client ID              | login         | `oauth.client.id`     |
+| OAuth client secert          | login         | `oauth.client.secret` |
```
Comment (elharo) [inline README.md]: secret

**Comment by elharo** [comment]
LGTM

**Pull request merged**